atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManagerImpl.java

```diff
@@ -114,6 +115,34 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
         return numTimesFailed > NUM_RETRIES;
     }
 
+    public <C extends PreCommitCondition, E extends Exception> ServiceWriteTransaction getWriteTransaction(C condition)
```

Comment:
I assume non-PoC version will add the methods to the API. This probably needs to be on the Serializable version.